hyperapp.d.ts

```diff
@@ -34,8 +34,8 @@ export type Children = VNode | string | number | null
  *
  * @memberOf [VDOM]
  */
-export function h<Attributes>(
-  nodeName: Component<Attributes> | string,
+export function h<Attributes, State = {}, Actions = {}>(
```

Comment:
@yuku-t Is it required to add State={}, Actions={} to `h< ...>`? The `h` function has never nothing to do with the current lazy component implementation.

Reply:
`nodeName` can contain a `Component` that require `State` and `Actions`

And both are optional so, I do not see any point against this.

Reply:
@Swizz The lazy component feature has really nothing to do with actual components — it's just the Vdom's ability to handle nodes of type function.

Reply:
You can give to `h` as first parameters a `Component` that can be lazy, right ?

Reply:
@Swizz You can give any function to `h` and `h` will call it. So, it's not like `h` will do anything special for lazy components.

Reply:
To remove `State` and `Actions`, we have to change the next line as:

```ts
nodeName: Component<Attributes, any, any> | string,
```

In this case, developers who directly use `h` (not via JSX) cannot specify state and actions. I think almost developers are using JSX, so it is ok to remove them. How do you think?

Reply:
So do we need to remove the `Component` type of `h` parameter to `Function` ?

Reply:
@yuku-t People using `h` will be calling components like `MyComponent(...)` and not like `h(MyComponent)`, so I think it's okay.

Reply:
@Swizz If we replace `Component` with `Function`, TypeScript compiler no longer checks direct `h` call:

![image](https://user-images.githubusercontent.com/96157/37837032-7fe92f02-2ef7-11e8-8275-db64fe7dfdcf.png)

so we shouldn't do that.

@jorgebucaran 🙆‍♂️

Reply:
done